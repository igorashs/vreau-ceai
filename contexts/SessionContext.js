import { createContext, useContext, useState } from 'react';

const SessionContext = createContext();

function SessionProvider({ children, session }) {
  const [curSession] = useState(session);

  return (
    <SessionContext.Provider value={curSession ?? {}}>
      {children}
    </SessionContext.Provider>
  );
}

function useSession() {
  const context = useContext(SessionContext);

  if (!context) {
    throw new Error('useSession must be used within a SessionProvider');
  }

  return context;
}

export { SessionProvider, useSession };
