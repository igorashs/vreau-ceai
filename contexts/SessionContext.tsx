import { createContext, useContext } from 'react';

type Session = {
  isAuth: boolean;
  user: {
    _id: string;
    name: string;
    isAdmin: boolean;
    isManager: boolean;
  };
  needRefresh: boolean;
};

const SessionContext = createContext<Session>({} as Session);

interface SessionProviderProps {
  session: any;
}

const SessionProvider = ({
  children,
  session,
}: React.PropsWithChildren<SessionProviderProps>) => {
  return (
    <SessionContext.Provider value={session}>
      {children}
    </SessionContext.Provider>
  );
};

const useSession = () => {
  const context = useContext(SessionContext);

  if (!context) {
    throw new Error('useSession must be used within a SessionProvider');
  }

  return context;
};

export { SessionProvider, useSession };
