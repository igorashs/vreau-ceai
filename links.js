import CartSvg from '@icons/cart.svg';
import LoginSvg from '@icons/login.svg';
import LogoutSvg from '@icons/logout.svg';
import PhoneSvg from '@icons/phone.svg';
import EmailSvg from '@icons/email.svg';
import PackageSvg from '@icons/package.svg';
import BriefcaseSvg from '@icons/briefcase.svg';

const linkFactory = ({
  path,
  text,
  label,
  Icon = null,
  accent = true,
  privatePath = false,
  managementPath = false
}) => ({
  path,
  text,
  label,
  Icon,
  accent,
  privatePath,
  managementPath
});

export const categoriesLink = linkFactory({
  path: '/categories',
  text: 'Ceai',
  label: 'Deschideți pagina cu categorii',
  accent: false
});

export const aboutLink = linkFactory({
  path: '/about',
  text: 'Despre',
  label: 'Deschideți pagina cu informații despre noi',
  accent: false
});

export const contactsLink = linkFactory({
  path: '/contacts',
  text: 'Contacte',
  label: 'Deschideți pagina cu contacte',
  accent: false
});

export const cartLink = linkFactory({
  path: '/cart',
  text: 'Coș',
  label: 'Deschideți pagina cu coșul personal',
  Icon: CartSvg,
  accent: false
});

export const loginLink = linkFactory({
  path: '/login',
  text: 'Conectați-vă',
  label: 'Deschideți pagina pentru conectare',
  Icon: LoginSvg
});

export const myOrdersLink = linkFactory({
  path: '/my-orders',
  text: 'Comenzile Mele',
  label: 'Deschideți pagina cu comenziile efectuate',
  Icon: PackageSvg,
  accent: false,
  privatePath: true
});

export const logoutLink = linkFactory({
  path: '/login',
  text: 'Deconectare',
  label: 'Deconectare din account',
  Icon: LogoutSvg,
  privatePath: true,
  accent: false
});

export const managementLink = linkFactory({
  path: '/management',
  text: 'Managment',
  label: 'Managment',
  Icon: BriefcaseSvg,
  privatePath: true,
  accent: true,
  managementPath: true
});

export const navLinks = [
  managementLink,
  categoriesLink,
  aboutLink,
  contactsLink,
  cartLink,
  loginLink,
  myOrdersLink,
  logoutLink
];

export const footerBlockLinksList = [
  {
    ...aboutLink,
    links: [
      linkFactory({
        path: '/about',
        text: 'vreau ceai',
        label: 'Deschideți pagina cu informații despre noi'
      })
    ]
  },
  {
    ...linkFactory({
      path: '/faq',
      text: 'FAQ',
      label: 'Deschideți pagina cu întrebări frecvente',
      accent: false
    }),
    links: [
      linkFactory({
        path: '/faq#cum-comand',
        text: 'cum comand',
        label: 'Deschideți pagina cu întrebarea cum comand'
      }),
      linkFactory({
        path: '/faq#cum-achit',
        text: 'cum achit',
        label: 'Deschideți pagina cu întrebarea cum achit'
      }),
      linkFactory({
        path: '/faq#livrare',
        text: 'cum are loc livrarea',
        label: 'Deschideți pagina cu întrebarea despre cum are loc livrarea'
      })
    ]
  },
  {
    ...contactsLink,
    links: [
      linkFactory({
        path: 'tel:062222222',
        text: '062222222',
        label: 'apelați nr de tel',
        Icon: PhoneSvg
      }),
      linkFactory({
        path: 'mailto:vreauceai@gmail.com',
        text: 'vreauceai@gmail.com',
        label: 'trimiteți email',
        Icon: EmailSvg
      })
    ]
  }
];
