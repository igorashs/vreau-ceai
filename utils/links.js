import CartSvg from '@/icons/cart.svg';
import LoginSvg from '@/icons/login.svg';
import PhoneSvg from '@/icons/phone.svg';
import EmailSvg from '@/icons/email.svg';
import PackageSvg from '@/icons/package.svg';
import BriefcaseSvg from '@/icons/briefcase.svg';

const linkFactory = ({
  href,
  text,
  label,
  Icon = null,
  accent = 'light',
  privatePath = false,
  managementPath = false,
  underline = false
}) => ({
  href,
  text,
  label,
  Icon,
  accent,
  privatePath,
  managementPath,
  underline
});

export const categoriesLink = linkFactory({
  href: '/categories',
  text: 'Ceai',
  label: 'Deschideți pagina cu categorii',
  accent: false
});

export const aboutLink = linkFactory({
  href: '/about',
  text: 'Despre',
  label: 'Deschideți pagina cu informații despre noi',
  accent: false
});

export const contactsLink = linkFactory({
  href: '/contacts',
  text: 'Contacte',
  label: 'Deschideți pagina cu contacte',
  accent: false
});

export const cartLink = linkFactory({
  href: '/cart',
  text: 'Coș',
  label: 'Deschideți pagina cu coșul personal',
  Icon: CartSvg,
  accent: false
});

export const loginLink = linkFactory({
  href: '/login',
  text: 'Conectați-vă',
  label: 'Deschideți pagina pentru conectare',
  Icon: LoginSvg
});

export const signupLink = linkFactory({
  href: '/signup',
  text: 'Creați un cont',
  label: 'Deschideți pagina pentru conectare'
});

export const myOrdersLink = linkFactory({
  href: '/my-orders',
  text: 'Comenzile Mele',
  label: 'Deschideți pagina cu comenziile efectuate',
  Icon: PackageSvg,
  accent: false,
  privatePath: true
});

export const managementLink = linkFactory({
  href: '/management',
  text: 'Managment',
  label: 'Managment',
  Icon: BriefcaseSvg,
  privatePath: true,
  managementPath: true
});

export const phoneLink = linkFactory({
  href: 'tel:062222222',
  text: '062222222',
  label: 'apelați nr de tel',
  Icon: PhoneSvg
});

export const emailLink = linkFactory({
  href: 'mailto:vreauceai@gmail.com',
  text: 'vreauceai@gmail.com',
  label: 'trimiteți email',
  Icon: EmailSvg
});

export const navLinks = {
  public: [categoriesLink, aboutLink, contactsLink, cartLink, loginLink],
  private: [categoriesLink, aboutLink, contactsLink, cartLink, myOrdersLink],
  management: [
    managementLink,
    categoriesLink,
    aboutLink,
    contactsLink,
    cartLink,
    myOrdersLink
  ]
};

export const managementLinks = {
  users: [
    linkFactory({
      href: '/management/users/update#form',
      text: 'Modificare Utilizator',
      accent: 'dark'
    })
  ],

  orders: [
    linkFactory({
      href: '/management/orders/all',
      text: 'Toate comenzile',
      accent: 'dark'
    }),
    linkFactory({
      href: '/management/orders/update',
      text: 'Modificare comandă',
      accent: 'dark'
    })
  ],

  products: [
    linkFactory({
      href: '/management/products/all',
      text: 'Toate produsele',
      accent: 'dark'
    }),
    linkFactory({
      href: '/management/products/update',
      text: 'Modificare produs',
      accent: 'dark'
    }),
    linkFactory({
      href: '/management/products/add',
      text: 'Adăugare produs',
      accent: 'dark'
    })
  ],

  categories: [
    linkFactory({
      href: '/management/categories/all',
      text: 'Toate categoriile',
      accent: 'dark'
    }),
    linkFactory({
      href: '/management/categories/update',
      text: 'Modificare categorie',
      accent: 'dark'
    }),
    linkFactory({
      href: '/management/categories/add',
      text: 'Adăugare categorie',
      accent: 'dark'
    })
  ]
};

export const footerBlockLinksList = [
  {
    ...aboutLink,
    links: [
      linkFactory({
        href: '/about',
        text: 'vreau ceai',
        label: 'Deschideți pagina cu informații despre noi'
      })
    ]
  },
  {
    ...linkFactory({
      href: '/faq',
      text: 'FAQ',
      label: 'Deschideți pagina cu întrebări frecvente',
      accent: false
    }),
    links: [
      linkFactory({
        href: '/faq#cum-comand',
        text: 'cum comand',
        label: 'Deschideți pagina cu întrebarea cum comand'
      }),
      linkFactory({
        href: '/faq#cum-achit',
        text: 'cum achit',
        label: 'Deschideți pagina cu întrebarea cum achit'
      }),
      linkFactory({
        href: '/faq#livrare',
        text: 'cum are loc livrarea',
        label: 'Deschideți pagina cu întrebarea despre cum are loc livrarea'
      })
    ]
  },
  {
    ...contactsLink,
    links: [{ ...phoneLink }, { ...emailLink }]
  }
];
