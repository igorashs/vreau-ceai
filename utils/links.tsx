import CartSvg from '@/icons/cart.svg';
import LoginSvg from '@/icons/login.svg';
import PhoneSvg from '@/icons/phone.svg';
import EmailSvg from '@/icons/email.svg';
import PackageSvg from '@/icons/package.svg';
import BriefcaseSvg from '@/icons/briefcase.svg';

type Link = {
  href: string;
  text: string;
  label?: string;
  Icon?: React.FC<React.SVGProps<SVGElement>> | null;
  accent?: 'light' | 'dark';
  privatePath?: boolean;
  managementPath?: boolean;
  underline?: boolean;
};

const linkFactory = ({
  href,
  text,
  label = '',
  Icon,
  accent,
  privatePath = false,
  managementPath = false,
  underline = false,
}: Link) => ({
  href,
  text,
  label,
  Icon,
  accent,
  privatePath,
  managementPath,
  underline,
});

export const categoriesLink = linkFactory({
  href: '/categories',
  text: 'Ceai',
  label: 'Deschideți pagina cu categorii',
});

export const aboutLink = linkFactory({
  href: '/about',
  text: 'Despre',
  label: 'Deschideți pagina cu informații despre noi',
});

export const contactsLink = linkFactory({
  href: '/contacts',
  text: 'Contacte',
  label: 'Deschideți pagina cu contacte',
});

export const cartLink = linkFactory({
  href: '/cart',
  text: 'Coș',
  label: 'Deschideți pagina cu coșul personal',
  Icon: CartSvg,
});

export const loginLink = linkFactory({
  href: '/login',
  text: 'Conectați-vă',
  label: 'Deschideți pagina pentru conectare',
  Icon: LoginSvg,
  accent: 'light',
});

export const signupLink = linkFactory({
  href: '/signup',
  text: 'Creați un cont',
  label: 'Deschideți pagina pentru conectare',
});

export const myOrdersLink = linkFactory({
  href: '/my-orders',
  text: 'Comenzile Mele',
  label: 'Deschideți pagina cu comenziile efectuate',
  Icon: PackageSvg,
  privatePath: true,
});

export const managementLink = linkFactory({
  href: '/management',
  text: 'Managment',
  label: 'Managment',
  accent: 'light',
  Icon: BriefcaseSvg,
  privatePath: true,
  managementPath: true,
});

export const phoneLink = linkFactory({
  href: 'tel:062222222',
  text: '062222222',
  label: 'apelați nr de tel',
  Icon: PhoneSvg,
  accent: 'light',
});

export const emailLink = linkFactory({
  href: 'mailto:vreauceai@gmail.com',
  text: 'vreauceai@gmail.com',
  label: 'trimiteți email',
  Icon: EmailSvg,
  accent: 'light',
});

export const navLinks = {
  public: [categoriesLink, aboutLink, contactsLink, cartLink, loginLink],
  private: [categoriesLink, aboutLink, contactsLink, cartLink, myOrdersLink],
  management: [
    managementLink,
    categoriesLink,
    aboutLink,
    contactsLink,
    cartLink,
    myOrdersLink,
  ],
};

export const managementLinks = {
  users: [
    linkFactory({
      href: '/management/users/update',
      text: 'Modificare Utilizator',
      accent: 'dark',
    }),
  ],

  orders: [
    linkFactory({
      href: '/management/orders',
      text: 'Toate comenzile',
      accent: 'dark',
    }),
    linkFactory({
      href: '/management/orders/update',
      text: 'Modificare comandă',
      accent: 'dark',
    }),
  ],

  products: [
    linkFactory({
      href: '/management/products',
      text: 'Toate produsele',
      accent: 'dark',
    }),
    linkFactory({
      href: '/management/products/update',
      text: 'Modificare produs',
      accent: 'dark',
    }),
    linkFactory({
      href: '/management/products/create',
      text: 'Adăugare produs',
      accent: 'dark',
    }),
  ],

  categories: [
    linkFactory({
      href: '/management/categories',
      text: 'Toate categoriile',
      accent: 'dark',
    }),
    linkFactory({
      href: '/management/categories/update',
      text: 'Modificare categorie',
      accent: 'dark',
    }),
    linkFactory({
      href: '/management/categories/create',
      text: 'Adăugare categorie',
      accent: 'dark',
    }),
  ],
};

export const footerBlockLinksList = [
  {
    ...aboutLink,
    links: [
      linkFactory({
        href: '/about',
        text: 'vreau ceai',
        label: 'Deschideți pagina cu informații despre noi',
        accent: 'light',
      }),
    ],
  },
  {
    ...linkFactory({
      href: '/faq',
      text: 'FAQ',
      label: 'Deschideți pagina cu întrebări frecvente',
    }),
    links: [
      linkFactory({
        href: '/faq#cum-comand',
        text: 'cum comand',
        label: 'Deschideți pagina cu întrebarea cum comand',
        accent: 'light',
      }),
      linkFactory({
        href: '/faq#cum-achit',
        text: 'cum achit',
        label: 'Deschideți pagina cu întrebarea cum achit',
        accent: 'light',
      }),
      linkFactory({
        href: '/faq#livrare',
        text: 'cum are loc livrarea',
        label: 'Deschideți pagina cu întrebarea despre cum are loc livrarea',
        accent: 'light',
      }),
    ],
  },
  {
    ...contactsLink,
    links: [{ ...phoneLink }, { ...emailLink }],
  },
];
