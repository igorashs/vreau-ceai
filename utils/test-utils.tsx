import { render, RenderOptions } from '@testing-library/react';
import GlobalStyle from 'GlobalStyle';

const Wrapper: React.FC = ({ children }) => (
  <>
    <GlobalStyle />
    {children}
  </>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
