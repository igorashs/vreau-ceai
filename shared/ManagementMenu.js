import styled from 'styled-components';
import { StyledLink } from './StyledLink';
import { DropDown } from './DropDown';
import { managementLinks } from '@/utils/links';

const Wrapper = styled.div`
  h4 {
    margin-bottom: var(--baseline);
  }

  h5 {
    margin-top: var(--baseline);
    margin-bottom: calc(var(--baseline) / 4);
  }
`;

const { users, orders, products, categories } = managementLinks;

export function ManagementMenu() {
  return (
    <Wrapper>
      <h4>Gestionare</h4>

      <h5>Users management</h5>
      <DropDown label="Utilizatori">
        {users.map((l) => (
          <li key={l.href}>
            <StyledLink {...l} />
          </li>
        ))}
      </DropDown>

      <h5>Shop management</h5>
      <DropDown label="Comenzi">
        {orders.map((l) => (
          <li key={l.href}>
            <StyledLink {...l} />
          </li>
        ))}
      </DropDown>

      <DropDown label="Produse">
        {products.map((l) => (
          <li key={l.href}>
            <StyledLink {...l} />
          </li>
        ))}
      </DropDown>

      <DropDown label="Categorii">
        {categories.map((l) => (
          <li key={l.href}>
            <StyledLink {...l} />
          </li>
        ))}
      </DropDown>
    </Wrapper>
  );
}
