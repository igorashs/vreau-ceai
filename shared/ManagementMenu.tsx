import styled from 'styled-components';
import { managementLinks } from '@/utils/links';
import { useSession } from 'contexts/SessionContext';
import StyledLink from './StyledLink';
import DropDownList from './DropDownList';

const Wrapper = styled.div`
  h4 {
    margin-bottom: var(--baseline);
  }

  h5 {
    margin-top: var(--baseline);
    margin-bottom: calc(var(--baseline) / 4);
  }
`;

const { users, orders, products, categories } = managementLinks;

const ManagementMenu = () => {
  const { user } = useSession();

  return (
    <Wrapper>
      <h4>Gestionare</h4>

      {user?.isAdmin && (
        <>
          <h5>Users management</h5>
          <DropDownList label="Utilizatori">
            {users.map((l) => (
              <li key={l.href}>
                <StyledLink {...l} />
              </li>
            ))}
          </DropDownList>
        </>
      )}

      {user?.isManager && (
        <>
          <h5>Shop management</h5>
          <DropDownList label="Comenzi">
            {orders.map((l) => (
              <li key={l.href}>
                <StyledLink {...l} />
              </li>
            ))}
          </DropDownList>

          <DropDownList label="Produse">
            {products.map((l) => (
              <li key={l.href}>
                <StyledLink {...l} />
              </li>
            ))}
          </DropDownList>

          <DropDownList label="Categorii">
            {categories.map((l) => (
              <li key={l.href}>
                <StyledLink {...l} />
              </li>
            ))}
          </DropDownList>
        </>
      )}
    </Wrapper>
  );
};

export default ManagementMenu;
