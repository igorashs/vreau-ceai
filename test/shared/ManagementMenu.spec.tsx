import { render } from '@/utils/test-utils';
import ManagementMenu from '@/shared/ManagementMenu';
import { SessionProvider } from 'contexts/SessionContext';
import { managementLinks } from '@/utils/links';

const { users, orders, products, categories } = managementLinks;

const session = {
  isAuth: true,
  user: {
    _id: 'user1',
    name: 'kekw',
    isAdmin: true,
    isManager: true,
  },
  needRefresh: false,
};

describe('ManagementMenu', () => {
  it('renders correctly', () => {
    const { container, getByText, getAllByRole } = render(
      <SessionProvider session={session}>
        <ManagementMenu />
      </SessionProvider>,
    );

    expect(getByText('Users management')).toBeVisible();
    expect(getByText('Shop management')).toBeVisible();
    expect(getByText(users[0].text)).toBeVisible();
    expect(getByText(orders[0].text)).toBeVisible();
    expect(getByText(products[0].text)).toBeVisible();
    expect(getByText(categories[0].text)).toBeVisible();
    expect(getAllByRole('listitem').length).toBe(
      Object.values(managementLinks).reduce(
        (count, links) => links.length + count,
        0,
      ),
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('when isAdmin is false', () => {
    it("doesn't render the admin links", () => {
      const { queryByText } = render(
        <SessionProvider
          session={{ ...session, user: { ...session.user, isAdmin: false } }}
        >
          <ManagementMenu />
        </SessionProvider>,
      );

      expect(queryByText('Users management')).not.toBeInTheDocument();
      expect(queryByText(users[0].text)).not.toBeInTheDocument();
    });
  });

  describe('when isManager is false', () => {
    it("doesn't render the manager links", () => {
      const { queryByText } = render(
        <SessionProvider
          session={{ ...session, user: { ...session.user, isManager: false } }}
        >
          <ManagementMenu />
        </SessionProvider>,
      );

      expect(queryByText('Shop management')).not.toBeInTheDocument();
      expect(queryByText(orders[0].text)).not.toBeInTheDocument();
      expect(queryByText(products[0].text)).not.toBeInTheDocument();
      expect(queryByText(categories[0].text)).not.toBeInTheDocument();
    });
  });
});
