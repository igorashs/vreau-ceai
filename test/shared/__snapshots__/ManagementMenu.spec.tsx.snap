// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManagementMenu renders correctly 1`] = `
.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 7px;
  padding: calc(var(--baseline) / 4) calc(var(--baseline) / 2);
  border-radius: 4px;
  border: 0;
  color: var(--text-light);
  background-color: var(--accent-dark);
  box-shadow: 1px 1px 2px #00000066;
  text-transform: lowercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  padding: 4px;
  box-shadow: none;
  padding: 0;
}

.c2:hover {
  background-color: var(--accent);
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 7px;
  color: var(--accent-text-dark);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover {
  -webkit-filter: brightness(0.7);
  filter: brightness(0.7);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: calc(var(--baseline) / 4) 0;
  border-bottom: 1px solid var(--layout);
}

.c1 p {
  font-weight: 500;
}

.c1 button {
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c3 {
  display: grid;
  -webkit-column-gap: var(--baseline);
  column-gap: var(--baseline);
  grid-template-columns: repeat(auto-fit,minmax(224px,1fr));
}

.c3 li {
  padding: calc(var(--baseline) / 4) 0;
  border-bottom: 1px solid var(--layout);
}

.c0 h4 {
  margin-bottom: var(--baseline);
}

.c0 h5 {
  margin-top: var(--baseline);
  margin-bottom: calc(var(--baseline) / 4);
}

<div
  class="c0"
>
  <h4>
    Gestionare
  </h4>
  <h5>
    Users management
  </h5>
  <div>
    <div
      class="c1"
    >
      <p>
        Utilizatori
      </p>
      <button
        class="c2"
      >
        <svg />
      </button>
    </div>
    <ul
      class="c3"
    >
      <li>
        <a
          aria-label=""
          class="c4"
          href="/management/users/update"
          rel=""
          target=""
        >
          Modificare Utilizator
        </a>
      </li>
    </ul>
  </div>
  <h5>
    Shop management
  </h5>
  <div>
    <div
      class="c1"
    >
      <p>
        Comenzi
      </p>
      <button
        class="c2"
      >
        <svg />
      </button>
    </div>
    <ul
      class="c3"
    >
      <li>
        <a
          aria-label=""
          class="c4"
          href="/management/orders"
          rel=""
          target=""
        >
          Toate comenzile
        </a>
      </li>
      <li>
        <a
          aria-label=""
          class="c4"
          href="/management/orders/update"
          rel=""
          target=""
        >
          Modificare comandă
        </a>
      </li>
    </ul>
  </div>
  <div>
    <div
      class="c1"
    >
      <p>
        Produse
      </p>
      <button
        class="c2"
      >
        <svg />
      </button>
    </div>
    <ul
      class="c3"
    >
      <li>
        <a
          aria-label=""
          class="c4"
          href="/management/products"
          rel=""
          target=""
        >
          Toate produsele
        </a>
      </li>
      <li>
        <a
          aria-label=""
          class="c4"
          href="/management/products/update"
          rel=""
          target=""
        >
          Modificare produs
        </a>
      </li>
      <li>
        <a
          aria-label=""
          class="c4"
          href="/management/products/create"
          rel=""
          target=""
        >
          Adăugare produs
        </a>
      </li>
    </ul>
  </div>
  <div>
    <div
      class="c1"
    >
      <p>
        Categorii
      </p>
      <button
        class="c2"
      >
        <svg />
      </button>
    </div>
    <ul
      class="c3"
    >
      <li>
        <a
          aria-label=""
          class="c4"
          href="/management/categories"
          rel=""
          target=""
        >
          Toate categoriile
        </a>
      </li>
      <li>
        <a
          aria-label=""
          class="c4"
          href="/management/categories/update"
          rel=""
          target=""
        >
          Modificare categorie
        </a>
      </li>
      <li>
        <a
          aria-label=""
          class="c4"
          href="/management/categories/create"
          rel=""
          target=""
        >
          Adăugare categorie
        </a>
      </li>
    </ul>
  </div>
</div>
`;
